Source/WebCore/loader/DocumentThreadableLoader.cpp

```diff
@@ -529,9 +529,7 @@ void DocumentThreadableLoader::didFail(std::optional<ResourceLoaderIdentifier>,
         clearResource();
 
         m_options.serviceWorkersMode = ServiceWorkersMode::None;
-        makeCrossOriginAccessRequestWithPreflight(WTFMove(m_bypassingPreflightForServiceWorkerRequest.value()));
-        ASSERT(m_bypassingPreflightForServiceWorkerRequest->isNull());
-        m_bypassingPreflightForServiceWorkerRequest = std::nullopt;
+        makeCrossOriginAccessRequestWithPreflight(*std::exchange(m_bypassingPreflightForServiceWorkerRequest, std::nullopt));
```

Comment:
@youennf , do you feel it is important to keep this assertion, or are you OK with this change?

Reply:
I think it is ok as is.
We could in theory, do:
```
auto request = *std::exchange(m_bypassingPreflightForServiceWorkerRequest, std::nullopt);
makeCrossOriginAccessRequestWithPreflight(WTFMove(request));
ASSERT(request.isNull();
```
Looking at the code, this ASSERT in incorrect anyway so your patch is good to go.